test/test_crud.py

```diff
@@ -20,14 +22,17 @@ def run_playbook(module, record=False, extra_vars=None):
         playbook_opts.extend(['--limit', 'tests'])
     if extra_vars:
         playbook_opts.extend(['--extra-vars', extra_vars])
-    # Dump recording parameters to json-file
+
+    # Dump recording parameters to json-file and pass its name by environment
     test_params = {'test_name': module, 'serial': 0, 'record_mode': record_mode}
-    with open('.vcr_params.json', 'w') as params_file:
-        json.dump(test_params, params_file)
-    # Call the playbook
-    cli = PlaybookCLI(playbook_opts)
-    cli.parse()
-    return cli.run()
+    with tempfile.NamedTemporaryFile('w', suffix='json', prefix='fam-vcr-') as params_file:
+        json.dump(test_params, params_file.file)
+        params_file.file.close()
```

Comment:
I think this won't interfere with the contextmanager but otherwise using ```flush()``` might be sufficient.

Reply:
I think flush or close doesn't matter. Since we were finished with writing, i thought we could as well close it.

Reply:
Wait, I thought you didn't have to explicitly call close() when you use a block like this.

Reply:
You don't, but due to buffering the file might not be written. Since the Ansible process wants to use this file, it might be empty. In this case the closing part also removes it.

Reply:
The contextmanager wraps the Tempfile_object. I am closing/flushing the embedded file object.
Hence `params.*file*.close()`.
Leaving the with context also deletes the tempfile which i think is nice behaviour.